src/Management/src/Endpoint/Middleware/EndpointMiddleware.cs

```diff
@@ -57,23 +57,68 @@ public async Task InvokeAsync(HttpContext context, RequestDelegate? next)
 
         if (ShouldInvoke(context.Request.Path))
         {
-            TResult result = await InvokeEndpointHandlerAsync(context, context.RequestAborted);
-            await WriteResponseAsync(result, context, context.RequestAborted);
+            if (!IsValidContentType(context.Request))
+            {
+                _logger.LogDebug("Content-Type header '{RequestContentType}' is not supported for this request.", context.Request.ContentType);
+                context.Response.StatusCode = (int)HttpStatusCode.UnsupportedMediaType;
+                await context.Response.WriteAsync($"Only the '{ContentType}' content type is supported.");
+            }
+            else if (!IsCompatibleAcceptHeader(context.Request))
+            {
+                _logger.LogDebug("Accept header '{AcceptType}' is not supported for this request.", context.Request.Headers.Accept.ToString());
+                context.Response.StatusCode = (int)HttpStatusCode.NotAcceptable;
+                await context.Response.WriteAsync($"Only the '{ContentType}' content type is supported.");
+            }
+            else
+            {
+                TResult result = await InvokeEndpointHandlerAsync(context, context.RequestAborted);
+                await WriteResponseAsync(result, context, context.RequestAborted);
+            }
         }
         else
         {
-            // Terminal middleware
             context.Response.StatusCode = (int)HttpStatusCode.NotFound;
         }
     }
 
+    private bool IsValidContentType(HttpRequest request)
+    {
+        if (request.ContentType == null)
+        {
+            return true;
+        }
+
+        // Media types are case-insensitive, according to https://stackoverflow.com/a/9842589.
+        return MediaTypeHeaderValue.TryParse(request.ContentType, out MediaTypeHeaderValue? headerValue) && headerValue.MatchesMediaType(ContentType);
+    }
+
+    private bool IsCompatibleAcceptHeader(HttpRequest request)
+    {
+        string[] acceptHeaderValues = request.Headers.GetCommaSeparatedValues("Accept");
+
+        if (acceptHeaderValues.Length == 0)
+        {
+            return true;
+        }
+
+        foreach (string acceptHeaderValue in acceptHeaderValues)
+        {
+            if (MediaTypeHeaderValue.TryParse(acceptHeaderValue, out MediaTypeHeaderValue? headerValue) && headerValue.MatchesMediaType(ContentType))
+            {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
     protected abstract Task<TResult> InvokeEndpointHandlerAsync(HttpContext context, CancellationToken cancellationToken);
 
     protected virtual async Task WriteResponseAsync(TResult result, HttpContext context, CancellationToken cancellationToken)
     {
         ArgumentNullException.ThrowIfNull(context);
 
-        context.HandleContentNegotiation(_logger);
+        context.Response.Headers.Append("Content-Type", ContentType);
```

Comment:
It would be more correct to set the content type _after_ checking whether `result` is null. I've tried to write a test, but we have no built-in actuators for that:
- Loggers endpoint has POST, but overrides this method
- Refresh endpoint has POST, but returns `{}` when `ReturnConfiguration` is set to `false`